src/js/elements/mapchip.js

```diff
@@ -119,4 +158,53 @@ export class MapChip extends Observable {
         this.updateMapChipText(label);
         this.showMapChip(args);
     }
+
+    setAddFilterButton() {
+        if (this.filter.allDropdowns.length >= this.groups.length * 2) {
+            $('a.mapping-options__add-filter').addClass('disabled');
```

Comment:
Could you extract the add button into its own widget? It can be in mapchip.js but should be a separate class